fea-rs/src/compile/feature_writer.rs

```diff
@@ -93,6 +99,58 @@ impl<'a> FeatureBuilder<'a> {
             .insert(class_name.into(), MarkClass { members });
     }
 
+    /// Define a mark class with a variable anchor.
+    ///
+    /// Roughly equivalent to `markClass glyph|class <anchor x y> @name{};`
+    /// [markClass](https://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#4.f)
+    /// with a variable anchor record.
+    pub fn add_mark_class_for_glyph(
+        &mut self,
+        gid: GlyphId,
+        default_anchor: (i16, i16),
+        class_name: impl Into<SmolStr>,
+        _anchor_variations: HashMap<Location, (i16, i16)>,
+    ) {
+        // TODO: an error type
+        eprintln!(
+            "TODO: {}",
+            format!(
+                "markClass {} <anchor {} {}> @{}; # TODO variable anchor",
+                gid,
+                default_anchor.0,
+                default_anchor.1,
+                class_name.into()
+            )
+        );
+    }
+
+    /// Setup a mark to base.
+    ///
+    /// Pseudo-fea:
+    ///
+    /// ```text
+    /// lookup name {
+    ///     pos base base_name <anchor x y> @markclass;
+    ///     // plus variations of anchor pos
+    /// }
+    ///
+    /// <https://adobe-type-tools.github.io/afdko/OpenTypeFeatureFileSpecification.html#6.d>
+    /// ```
+    pub fn add_mark_base_pos(
+        &mut self,
+        lookup_name: impl Into<SmolStr>,
+        base_name: impl Into<SmolStr>,
+        class_name: impl Into<SmolStr>,
+        default_anchor: (i16, i16),
+        _anchor_variations: HashMap<Location, (i16, i16)>,
```

Comment:
Here I'm a bit more sanguine. Using lookup names adds another layer of misdirection; I would prefer to have the caller construct lookups directly and then use the provided `LookupId` type to add lookups to a feature. My rationale here is that the whole idea of named lookups is a convenience in the FEA language, but is just a less efficient alias for a numerical ID, which we are going to need to create anyway.

Something I'm taking away from this, though, is that I think overall it would be nice if the `ValueRecord` and `AnchorTable` types used in fea-rs more naturally supported the inclusion of variation data, so I'll look into that.

Reply:
No objection to passing ids rather than names as long as we make acquiring an id easy. For example, perhaps here there need not be anything at all for lookup, I want a mark base, and the builder will generate a lookup to hold it.